lib/util/exec.ts

```diff
@@ -19,14 +22,41 @@ export interface ExecResult {
   stderr: string;
 }
 
-export function exec(cmd: string, options?: ExecOptions): Promise<ExecResult> {
+export function exec(cmd: string, opts: ExecOptions = {}): Promise<ExecResult> {
+  const { env: customEnv = {}, envVars = [], docker, cwd } = opts;
+
   let pExecCommand = cmd;
-  const pExecOptions = { ...options, encoding: 'utf-8' };
+  const pExecOptions = {
+    encoding: 'utf-8',
+    ...opts,
+  };
+  delete pExecOptions.envVars;
+  delete pExecOptions.docker;
+
+  const processEnv = getChildProcessEnv(envVars);
+  const isCustomEnv = !!opts.env || typeof opts.envVars !== 'undefined';
```

Comment:
Different trythiness semantics made first versions of my code work smoothly with existing tests/mocks, no there is no need for this.

Custom env is constructed if we (1) pass `env` option explicitly or (2) we have some additional vars to inherit from current `process.env`. Otherwise, we're using invironment obtained via `getChildProcessEnv(...)` call.

Reply:
In what cases do we require this custom env logic and which do we not? I'm still trying to understand it.

Reply:
https://github.com/renovatebot/renovate/blob/master/lib/manager/gomod/artifacts.ts#L54

After refactor, I'll pass `{ ..., env: { GCO_ENABLED: '0'}, ... }` as an exec option